Import unsafeCoerce# from GHC.Exts, not GHC.Base

`primitive` (both version 0.7.0.0 and on the `master` branch) fail to compile on GHC HEAD (8.11) after [this commit](https://gitlab.haskell.org/ghc/ghc/commit/74ad75e87317196c600dfabc61aee1b87d95c214):

```
[ 1 of 13] Compiling Control.Monad.Primitive ( Control/Monad/Primitive.hs, dist/build/Control/Monad/Primitive.o, dist/build/Control/Monad/Primitive.dyn_o )

Control/Monad/Primitive.hs:27:21: error:
    Module ‘GHC.Base’ does not export ‘unsafeCoerce#’
   |
27 | import GHC.Base   ( unsafeCoerce#, realWorld# )
   |                     ^^^^^^^^^^^^^
```

The reason is that `unsafeCoerce#` was moved from `GHC.Prim` (which `GHC.Base` reexports) to `Unsafe.Coerce`. Luckily, there is a backwards-compatible way to fix this: just import `unsafeCoerce#` from `GHC.Exts` instead, which has better stability API-wise.

More generally, I wonder if we should switch all `GHC.Base` imports to `GHC.Exts` (_à la_ #197) since `GHC.Base` has proven to be volatile like `GHC.Prim`.

--- a/Control/Monad/Primitive.hs
+++ b/Control/Monad/Primitive.hs
@@ -23,18 +23,9 @@ module Control.Monad.Primitive (
   touch, evalPrim, unsafeInterleave, unsafeDupableInterleave, noDuplicate
 ) where
 
-import GHC.Exts   ( State#, RealWorld, noDuplicate#, touch# )
-import GHC.Base   ( unsafeCoerce#, realWorld# )
-#if MIN_VERSION_base(4,4,0)
-import GHC.Base   ( seq# )
-#else
-import Control.Exception (evaluate)
-#endif
-#if MIN_VERSION_base(4,2,0)
+import GHC.Exts   ( State#, RealWorld, noDuplicate#, touch#
+                  , unsafeCoerce#, realWorld#, seq# )
 import GHC.IO     ( IO(..) )
-#else
-import GHC.IOBase ( IO(..) )
-#endif
 import GHC.ST     ( ST(..) )
 
 import Control.Monad.Trans.Class (lift)

--- a/changelog.md
+++ b/changelog.md
@@ -2,6 +2,8 @@
 
   * Add `Eq` instances for `MutableByteArray` and `MutablePrimArray`.
 
+  * Allow building with GHC 8.12.
+
 ## Changes in version 0.7.0.0
 
   * Remove `Addr` data type, lifted code should use `Ptr a` now

